storage/storage.go

```diff
@@ -241,6 +245,26 @@ type RefreshToken struct {
 	Nonce string
 }
 
+// RefreshTokenRef is a reference object that contains metadata about refresh tokens.
+type RefreshTokenRef struct {
+	ID        string
+	CreatedAt time.Time
+	LastUsed  time.Time
```

Comment:
One `OfflineSession` per user (maybe even rename it `OfflineSessions`?). But the `RefreshTokenRef` needs to contain a client ID. The steps for creating a new refresh token should be:

1. Get an existing OfflineSisson(s) object.
2. Is there a refresh token for the client currently requesting the new refresh token?
3. If so remove that refresh token from the list, add a new one.